File: src/main/java/com/hepdd/gtmthings/client/ClientUtil.java
package com.hepdd.gtmthings.client;

import net.minecraft.client.Minecraft;
import net.minecraft.client.multiplayer.ClientLevel;
import net.minecraft.client.renderer.ItemModelShaper;
import net.minecraft.client.renderer.MultiBufferSource;
import net.minecraft.client.renderer.entity.ItemRenderer;
import net.minecraft.client.resources.model.BakedModel;
import net.minecraft.world.entity.LivingEntity;
import net.minecraft.world.item.ItemDisplayContext;
import net.minecraft.world.item.ItemStack;
import net.minecraftforge.api.distmarker.Dist;
import net.minecraftforge.api.distmarker.OnlyIn;

import com.lowdragmc.lowdraglib.client.renderer.IItemRendererProvider;
import com.mojang.blaze3d.vertex.PoseStack;

@OnlyIn(Dist.CLIENT)
public final class ClientUtil {

    public static ItemRenderer getItemRenderer() {
        return Minecraft.getInstance().getItemRenderer();
    }

    public static BakedModel getVanillaModel(ItemStack stack, ClientLevel level, LivingEntity entity) {
        ItemModelShaper shaper = getItemRenderer().getItemModelShaper();
        BakedModel model = shaper.getItemModel(stack.getItem());
        if (model != null) {
            BakedModel bakedmodel = model.getOverrides().resolve(model, stack, level, entity, 0);
            if (bakedmodel != null) return bakedmodel;
        }
        return shaper.getModelManager().getMissingModel();
    }

    public static void vanillaRender(ItemStack stack, ItemDisplayContext transformType, boolean leftHand, PoseStack poseStack, MultiBufferSource buffer, int combinedLight, int combinedOverlay, BakedModel model) {
        IItemRendererProvider.disabled.set(true);
        getItemRenderer().render(stack, transformType, leftHand, poseStack, buffer, combinedLight, combinedOverlay, model);
        IItemRendererProvider.disabled.set(false);
    }
}

File: src/main/java/com/hepdd/gtmthings/client/ClientUtil.kt
package com.hepdd.gtmthings.client

import net.minecraft.client.Minecraft
import net.minecraft.client.multiplayer.ClientLevel
import net.minecraft.client.renderer.MultiBufferSource
import net.minecraft.client.renderer.entity.ItemRenderer
import net.minecraft.client.resources.model.BakedModel
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.item.ItemDisplayContext
import net.minecraft.world.item.ItemStack
import net.minecraftforge.api.distmarker.Dist
import net.minecraftforge.api.distmarker.OnlyIn

import com.lowdragmc.lowdraglib.client.renderer.IItemRendererProvider
import com.mojang.blaze3d.vertex.PoseStack

@OnlyIn(Dist.CLIENT)
class ClientUtil {
    companion object {

        @JvmStatic
        fun getItemRenderer(): ItemRenderer = Minecraft.getInstance().itemRenderer

        @JvmStatic
        fun getVanillaModel(stack: ItemStack, level: ClientLevel?, entity: LivingEntity?): BakedModel {
            val shaper = getItemRenderer().itemModelShaper
            val model = shaper.getItemModel(stack.item)
            if (model != null) {
                val bakedmodel = model.overrides.resolve(model, stack, level, entity, 0)
                if (bakedmodel != null) return bakedmodel
            }
            return shaper.modelManager.missingModel
        }

        @JvmStatic
        fun vanillaRender(stack: ItemStack, transformType: ItemDisplayContext, leftHand: Boolean, poseStack: PoseStack, buffer: MultiBufferSource, combinedLight: Int, combinedOverlay: Int, model: BakedModel) {
            IItemRendererProvider.disabled.set(true)
            getItemRenderer().render(
                stack,
                transformType,
                leftHand,
                poseStack,
                buffer,
                combinedLight,
                combinedOverlay,
                model,
            )
            IItemRendererProvider.disabled.set(false)
        }
    }
}
